pyabc/visualization/data_plot.py

```diff
@@ -0,0 +1,70 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def plot_data(obs_data: dict, sim_data: dict, key=None):
+    fig = plt.figure()
+    plot_index = 1
+    # check if user specify a specific key to be printed
+    if key is not None:
+        obs_data = {key: obs_data[key]}
+        sim_data = {key: sim_data[key]}
+    if len(obs_data) <= 3:
+        plot_row_size = 1
+        plot_col_size = len(obs_data)
+    elif len(obs_data) <= 6:
+        plot_row_size = 2
+        plot_col_size = 2
+    elif len(obs_data) <= 9:
+        plot_row_size = 3
+        plot_col_size = 3
+    elif len(obs_data) <= 16:
+        plot_row_size = 4
+        plot_col_size = 4
+    else:
+        print("Data length should be equal or less than 16 ")
+        return -1
```

Comment:
to my knowledge, return codes are uncommon in python. I would just raise the warning and then ``return``.
Please also use logger.warn (see how to use a logger here: https://github.com/ICB-DCM/pyABC/blob/master/pyabc/distance/distance.py#L12) instead of all print commands.